drivers/Ticker.h

```diff
@@ -76,21 +76,12 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
 
     /** Attach a function to be called by the Ticker, specifying the interval in seconds
      *
-     *  The method forwards its arguments to attach_us() rather than copying them which
-     *  may not be trivial depending on the callback copied.
-     *  The function is forcibly inlined to not use floating-point operations. This is
-     *  possible given attach_us() expects an integer value for the callback interval.
      *  @param func pointer to the function to be called
      *  @param t the time between calls in seconds
      */
-#if defined(__ICCARM__)
-    MBED_FORCEINLINE template <typename F>
-#else
-    template <typename F> MBED_FORCEINLINE
-#endif
-    void attach(F &&func, float t)
+    MBED_FORCEINLINE void attach(Callback<void()> func, float t)
     {
-        attach_us(std::forward<F>(func), t * 1000000.0f);
+        attach_us(func, (int)(t * 1000000));
```

Comment:
This breaks the API for anything >4000s. Is it the limiting to 32-bit that is making the difference for your test?

And that test is a non-constant use, I gather? Someone passing a non-constant integer to this?

If trying to avoid float use, you should fix the calling code to not use floats, not break this API.

Reply:
Changing the calling code as follows without making any changes to `Ticker.h` does not remove float symbols.
```C
void NFCController::scheduler_process(bool hw_interrupt)
{
    _timeout.detach(); // Cancel timeout - if it triggers, it's ok as we'll have an "early" iteration which will likely be a no-op

    // Process stack events
    uint32_t timeout_ms = nfc_scheduler_iteration(_scheduler, hw_interrupt ? EVENT_HW_INTERRUPT : EVENT_NONE);

    _timeout.attach(
        callback(this, &NFCController::on_timeout),
        timeout_ms/1000 // convert milliseconds to seconds <=====
    );
}
```

Reply:
Well, no, because you're still using the `attach(float)`.

That should be

    _timeout.attach_us(cb(blah), timeout_ms * (us_timestamp_t) 1000);

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/42bf8132c5557445bd38ce0927cbc26e9189ee83..ec2c34176762680af3b45ddc1f0673e71f635ba2).